src/ensembl/src/content/app/entity-viewer/state/general/entityViewerGeneralSlice.ts

```diff
@@ -90,6 +91,9 @@ export const setDataFromUrl =
           })
         );
       }
+      if (activeGenomeId === genomeIdFromUrl && entityId !== activeEntityId) {
```

Comment:
Is the first check necessary? If the entityId has changed we clear filters.

Reply:
@jyothishnt it is necessary. The reason for this is this scenario:
Human BRCA2, you apply filter -> switch to another species -> come back to human again
The filter should still be applied and the check make sure it does because when switching the species it was reloading the component and hence losing the filters.

Reply:
got it.. thanks